code/to_bsdd.py

```diff
@@ -404,30 +395,81 @@ def parents(k):
     #     return has_child_
 
     def should_include(k, v):
-        # PREVIOUSLY return ("IfcProduct" in parents(k)) or has_child("IfcProduct")(k) or child_or_self_has_psets(k) 
+        # PREVIOUSLY return ("IfcProduct" in parents(k)) or has_child("IfcProduct")(k) or
+        # child_or_self_has_psets(k)
         # but decided to widen to also include all non-products that have psets, like IfcActor.
-        # return ("IfcRoot" in parents(k)) or ("IfcMaterialDefinition" in parents(k)) or ("IfcProfileDef" in parents(k)) or child_or_self_has_psets(k)
-        # Now skipping relations ('IfcRelAssociatesClassification'), types ('IfcWallType'), property definitions ('IfcPropertySingleValue'), Resources and some other abstract concepts. 
-        x1 = k.startswith(('IfcRel','IfcProperty','IfcProperty','IfcQuantity','IfcConnection','IfcCartesian','IfcMaterial'))
-        x2 = k.endswith(('Relationship','Type','Usage','Property','Template','Resource','Select','Measure','Condition','ProfileDef','Value','Property','Quantity','Curve','Number','Reference','Information','Solid'))
-        x3 = k.endswith('Unit') and not k == 'IfcProtectiveDeviceTrippingUnit'
-        x4 = k.startswith('IfcAlignment') and not k == 'IfcAlignment'
-        x5 = k in EXCLUDED_ENTITIES
-        x6 = (k != "IfcObjectDefinition" and k.endswith("Definition"))
-        x7 = any(x in parents(k) for x in ["IfcPropertyAbstraction","IfcConstraint","IfcRepresentationItem","IfcPresentationItem","IfcPresentationStyle","IfcTypeObject","IfcExternalReference","IfcStructuralLoadOrResult",""])
+        # return ("IfcRoot" in parents(k)) or ("IfcMaterialDefinition" in parents(k)) or
+        # ("IfcProfileDef" in parents(k)) or child_or_self_has_psets(k)
+        # Now skipping relations ('IfcRelAssociatesClassification'), types ('IfcWallType'),
+        # property definitions ('IfcPropertySingleValue'), Resources and other abstract concepts.
+        x1 = k.startswith(
+            (
+                "IfcRel",
+                "IfcProperty",
+                "IfcProperty",
+                "IfcQuantity",
+                "IfcConnection",
+                "IfcCartesian",
+                "IfcMaterial",
+            )
+        )
+        x2 = k.endswith(
+            (
+                "Relationship",
+                "Type",
+                "Usage",
+                "Property",
+                "Template",
+                "Resource",
+                "Select",
+                "Measure",
+                "Condition",
+                "ProfileDef",
+                "Value",
+                "Property",
+                "Quantity",
+                "Curve",
+                "Number",
+                "Reference",
+                "Information",
+                "Solid",
+            )
+        )
+        x3 = k.endswith("Unit") and not k == "IfcProtectiveDeviceTrippingUnit"
```

Comment:
Particularly this one is rather brittle. We better do this while incorporating the actual inheritance hierarchy.

Reply:
Yes, that was an ugly script I did back then, and I have low on my to-do list to improve it one day. I want to be more precise in what classes make and what doesn't make it to bSDD. But I didn't want to change this in this PR - the only change here is that I extracted the definition splitter into a separate script, as well as moved the long lists of words to JSON files.